bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -150,65 +148,76 @@ public static String executeUrl(String httpMethod, String url, Properties httpHe
             String contentType, int timeout, String proxyHost, Integer proxyPort, String proxyUser,
             String proxyPassword, String nonProxyHosts) {
 
-        HttpClient client = new HttpClient();
+        startHttpClient(client);
 
         // only configure a proxy if a host is provided
         if (StringUtils.isNotBlank(proxyHost) && proxyPort != null && shouldUseProxy(url, nonProxyHosts)) {
-            client.getHostConfiguration().setProxy(proxyHost, proxyPort);
-            if (StringUtils.isNotBlank(proxyUser)) {
-                client.getState().setProxyCredentials(AuthScope.ANY,
-                        new UsernamePasswordCredentials(proxyUser, proxyPassword));
+            AuthenticationStore authStore = client.getAuthenticationStore();
+            ProxyConfiguration proxyConfig = client.getProxyConfiguration();
+
+            // clear proxy list from previous requests
+            List<Proxy> proxies = proxyConfig.getProxies();
+            if (!proxies.isEmpty()) {
+                proxies.clear();
             }
+
+            HttpProxy proxy = new HttpProxy(proxyHost, proxyPort);
+            proxies.add(proxy);
+
+            authStore.addAuthentication(new BasicAuthentication(proxy.getURI(), null, proxyUser, proxyPassword));
```

Comment:
I had read this as well, but noticed that BasicAuthentication.BasicResult wasn't available in our version (might be a very recent addition).
In any case, adding the header is imho much better, since it is stateless and we do not have to deal with removing it from the auth store again (and dealing with all associated concurrency issues). For the proxy, which is a global setting in Java, it is in contrast imho no problem to put its credentials in the auth store.